db/sg_replicate_cfg.go

```diff
@@ -244,12 +250,24 @@ func (rc *ReplicationConfig) Upsert(c *ReplicationUpsertConfig) {
 		rc.Remote = *c.Remote
 	}
 
-	if c.Username != nil {
-		rc.Username = *c.Username
+	if c.Username != nil || c.Password != nil {
+		base.Warnf(`Deprecation notice: replication config fields "username" and "password" are now "remote_username" and "remote_password" respectively`)
+		if c.Username != nil {
+			rc.RemoteUsername = *c.Username
+		}
```

Comment:
Doing this mapping here in Upsert from the old to the new option breaks backwards compatbility when just loading existing replication configurations.

We don't have test coverage to catch that case given we're always writing configs through Upsert the REST API, so we'll need to think of a way to test that.